### PR TITLE
Add LL/TIM/ADV/BV-03-C, LL/TIM/ADV/BV-04-C, LL/TIM/ADV/BV-05-C, LL/TIM/ADV/BV-07-C

### PR DESCRIPTION
**Pull request by Tronil**
Added the four (very similar test cases)

Added a findLast() convenience function to Packets for finding the last
packet matching the given packet_filter

Signed-off-by: Troels Nilsson [trnn@demant.com](mailto:trnn@demant.com)

### CODE REVIEW

File: src/tests/ll_verification.py [hunk 1]
```diff
@@ -8465,6 +8465,153 @@ def ll_ist_cen_bv_03_c(transport, upper_tester, lower_tester, trace):
...
+                    # Note: Packet air length is: pre-amble + AA + header + payload + CRC
+                    airTime = ((2 if auxAdvIndPacket.phy == '2M' else 1) + 4 + 2 + len(auxAdvIndPacket) + 3)*8/(2 if auxAdvIndPacket.phy == '2M' else 1)
+                    # Calculate expected (last possible) end time of the coming AUX_ADV_IND packet
+                    auxAdvIndEndTs = int(lastPacket.ts + offsetEnd + airTime)
+                    break
+                # Don't have the needed packets yet or the last packet was not an ADV_EXT_IND; Wait a little and try again
+                transport.wait(1)
+            if auxAdvIndEndTs + 1 <= transport.get_last_t():
+                # Always wait at least 1 us to avoid deadlocks
+                auxAdvIndEndTs = transport.get_last_t() + 1
```
Comment (wopu-ot) [inline src/tests/ll_verification.py:8528]: Wouldn't this "+1" and the "+1" in wait_until_t mean that this waits at least 2us?

File: src/components/dump.py [hunk 2]
```diff
@@ -615,7 +615,24 @@ def find(self, packet_type=None):
...
+    def findLast(self, packet_filter=()):
+        try:
+            iter(packet_filter)
+        except TypeError:
+            packet_filter = (packet_filter,)
+        else:
+            packet_filter = packet_filter
+
+        # Append new packets
+        for dump in self.__dumps.fetch():
```
Comment (wopu-ot) [inline src/components/dump.py:628]: Why is this function not fetching in a loop like the `fetch` function?

File: src/tests/ll_verification.py (hunk 1 above)
Comment (Tronil) [reply src/tests/ll_verification.py:8528]: True - I opted for the safe and readable number; anyway, waiting 2 us should be fine as well :-)

File: src/components/dump.py (hunk 2 above)
Comment (Tronil) [reply src/components/dump.py:628]: Performance mostly. Looping through all the packages several hundred times just to get the last packet had a noticeable impact on run-time.
In addition, it is also slightly easier to use than having a loop.

File: src/tests/ll_verification.py (hunk 1 above)
Comment (wopu-ot) [reply src/tests/ll_verification.py:8528]: I am sure it works fine, but the comment is misleading then. Maybe just doing `transport.wait_until_t(max(auxAdvIndEndTs, transport.get_last_t()) + 1)` below would be simpler?

Comment (Tronil) [reply src/tests/ll_verification.py:8528]: Yeah that would be cleaner - I'll change it to use that instead

**Review by wopu-ot — APPROVED** [review]

**Pull request merged**